src/main/scala/rocket/ICache.scala

```diff
@@ -436,6 +443,19 @@ class ICacheModule(outer: ICache) extends LazyModuleImp(outer)
     ccover(!refill_valid && (tl_out.d.fire() && !refill_one_beat), "PREFETCH_D_AFTER_MISS_D", "I$ prefetch resolves after miss")
     ccover(tl_out.a.fire() && hint_outstanding, "PREFETCH_D_AFTER_MISS_A", "I$ prefetch resolves after second miss")
   }
+  // Drive APROT information
+  tl_out.a.bits.user.lift(AMBAProt).foreach { x =>
+    val user_bit_cacheable = edge_out.manager.supportsAcquireTFast(refill_paddr, lgCacheBlockBytes.U)
+
+    x.privileged  := io.privileged || user_bit_cacheable    // privileged if machine mode or memory port
```

Comment:
oh i see the comment here. so we are assuming everythign cacheable is going to memory port? If not this comment is a bit misleading

Reply:
Yes, currently we are assuming any cacheable thing is going to the memory port.  During a short discussion that I had with @hcook he mentioned that it's possible this may change in the future if we add other types of ports.

Reply:
It seems it should be the opposite for safety, i.e. if you cannot figure out the exact privilege level, you should assume the lowest level of privilege.

Reply:
I don't understand ARM's reasoning behind their decision, but we are making this change to be more compatible with ARM behavior.  So I think for that reason we should follow the same thing.

Reply:
I am confused. Since we cache everything on fetch, why is privileged not just tied to true?

Reply:
@terpstra, @aswaterman posed a similar question but was unsure.